### DISCUSSION TITLE
MySpeed 1.1.0 Redesign: What are your thoughts?

### DISCUSSION THREAD

**Discussion by gnmyt** [gnmyt/MySpeed#744] [Q&A]
With the new 1.1.0 version, I want to address some issues I've had with the UI and want to gather some feedback from you to see what you think about it. Please note that these changes are not final and are subject to change.

## View switching

You might already know that there are two different views in MySpeed. The "test overview" and "test statistics", which show the tests either in a list or as "widgets".

Currently, you need to go into the dropdown and choose "Change view" to switch the new view. I personally switch the view very often so I want to introduce a page switching system at the top of the page instead of one hidden in the dropdown.

![Landing Page](https://github.com/user-attachments/assets/0a06bbf3-e4d7-464b-ab11-9269bd70a539)

## Container transparency

As you might have already seen in the screenshot, I switched the container background to a darker color instead of a transparent one. It took me a while to get used to but overall I think this makes the containers cleaner. I also think that it fits better to the rest of the page.

## Scroll to load

The next problem is the list overview itself. The current state is that you can only see tests up to 2 days here. If you wanted to go older than that, you would need to view averages. Also, if you chose an schedule like every minute, this gets really inefficient since it loads so many tests that the page begins to lag.

To fix this, I want to only load the tests that are visible. If you would want to load more than that, just scroll down. It will then automatically load more results.

I also want to add a little tooltip at the top that shows the current date when scrolling since it could get confusing which day the tests belong to when you scrolled too much.

![Scrolling](https://github.com/user-attachments/assets/4a1b531b-68a0-42d7-8c34-1d49ca1661bb)

## The statistics page

I also want to introduce some changes in the Statistics page. First of all, the widgets now also have the background like in the test overview. I wanted to completely get rid of the "Set period" dropdown item and rather have a time chooser at the top left.

![Statistics](https://github.com/user-attachments/assets/5d17b4c6-9798-42b6-816e-46b65378a622)

I also want to introduce an "Edit mode", which allows you to add, delete and change the order of the widgets on the page.

To make the statistics more useful, I decided to make the widgets go fullscreen when clicking them. This could look like this:

![StatisticsD](https://github.com/user-attachments/assets/752d1adb-8b18-4752-b790-01f940644512)

## Analyzer

As you might have seen at the top of the page, there will be a new "Analyzer" page. However this will be another discussion since I want to focus this thread on the new page design.

**Comment by CrazyWolf13** [comment]
Looks absolutely amazing!!

Just out of curiosity, what is this UI written in?

Is there some sort of Prometheus statistics planned?

**Reply by gnmyt** [reply]
Thank you. The UI of MySpeed is written in [React](https://react.dev/). However, this concept has been made in [Figma](https://figma.com/).

MySpeed also already has an Prometheus endpoint. Currently, you can find documentation about it in PR #687, but I will make sure to add a more detailed documentation on [https://docs.myspeed.dev](https://docs.myspeed.dev) soon.

**Reply by CrazyWolf13** [reply]
@gnmyt Oh I see, thanks yeah absolutely love the design!!


Oh that's awesome, thanks!

**Comment by adm2k** [comment]
These look like excellent updates!

I do have a question, is there a way to get a simple notification (email) when download speed drops below a specific level and/or ping is above a specified level?

Many thanks!

**Reply by CrazyWolf13** [reply]
That would be indeed a very nice feature request, possibly this could be some sort of general protocol like shoutrr to support ntfy.sh ?

**Reply by gnmyt** [reply]
Thanks. Support for ntfy and email is already planned (#711). Notifications about a certain threshold will also be part of the "Analyzer" update, which basically tracks network speeds for a certain amount of time and displays them on a separate page. However, this will be a different discussion.

**Comment by CrazyWolf13** [comment]
Is it possible to set custom time durations?
Like cronjob schedule or so?

As for me personally three times a days is a bit too much. I'd prefer like once a day or so.

**Reply by gnmyt** [reply]
Yes, you can set a custom duration with cron expressions. Just go into the settings > Set frequency > Set manually. If you need help with cron expressions, you can use [this website](https://crontab.guru/).

**Reply by CrazyWolf13** [reply]
Oh that's awesome, sorry I did not see that.

Thanks!

**Comment by mguilherme** [comment]
that looks awesome, I'm trying a new internet provider and this is really awesome, that would be great if you could add a light mode, I created an issue for that: https://github.com/gnmyt/myspeed/issues/795

**Reply by Kristoffeh** [reply]
This is the first time in my lifetime i have seen someone request light mode.

**Comment by Kristoffeh** [comment]
While i am not much of a fan when it comes to radial graphs, primarily because they need interaction in order to understand what they're saying. Regardless, this does look really good.

**Reply by burgerga** [reply]
See https://github.com/gnmyt/myspeed/issues/768, already solved (came here to ask the same thing :P)

**Reply by Kristoffeh** [reply]
Oh cool, thank you @burgerga

**Comment by MitoG** [comment]
Just a small note on the Statistics page.

I feel the element `Failed Tests` should be renamed since it doesn't only display failed tests but all test results, which makes it a bit confusing on a quick glance.

It's not really an issue with the redesign in on itself since it's also named that way in the current design but just my 2 cents

**Comment by burgerga** [comment]
Do you have an indication on when you want to release it? (the previous release is from May, and it seems you have been quite active in the mean time ;) )

**Comment by panzer-arc** [comment]
Please update the docs to include that the prometheus metrics port is the same as the server port. In my experience, most software uses a separate port for this.

Here is a translation of the post to English:

🔥 Prometheus endpoint added  
MySpeed now includes a Prometheus endpoint as requested. This will always display the data from the last test. The endpoint shows the following values: Download, Upload, Ping, Time, Current Server. This closes #515  

Here's a sample configuration for MySpeed in prometheus.yml:  

  - job_name: 'myspeed'  
    scrape_interval: 1m # Adjust based on test frequency  
    scrape_timeout: 8s  
    metrics_path: /api/prometheus/metrics  
    basic_auth:  
      username: prometheus  
      password: your-password # Only required if a password is set in the web interface  
    static_configs:  
      - targets: ['localhost:5216']  
It's important to adjust the scrape_interval. If you know you'll only run automated tests scheduled hourly, setting this to 1h is sufficient. This saves bandwidth and avoids unnecessary duplicates. For frequent manual tests where you want to ensure they're displayed, you can vary this setting.

**Comment by constgen** [comment]
It is better

**Comment by sdi72** [comment]
add widget to Homepage lxc
